src/Microsoft.OData.Core/Json/NonIndentedTextWriter.cs

```diff
@@ -41,11 +52,30 @@ public override void Write(char value)
             this.writer.Write(value);
         }
 
+        /// <summary>
+        /// Writes the given char value to the underlying writer asynchronously.
+        /// </summary>
+        /// <param name="value">Char value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        public override async Task WriteAsync(char value)
+        {
+            await this.writer.WriteAsync(value).ConfigureAwait(false);
+        }
+
         /// <summary>
         /// Writes a new line.
         /// </summary>
         public override void WriteLine()
         {
         }
+
+        /// <summary>
+        /// Writes a new line asynchronously.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        public override async Task WriteLineAsync()
+        {
+            await TaskUtils.CompletedTask.ConfigureAwait(false);

```

Comment:
Doesn't this need to call WriteLineAsync on the underling writer?

Reply:
@mikepizzo For the `NonIndentedTextWriter`, we don't really apply newlines and indentations. Kindly take a look at the equivalent [`WriteLine`](https://github.com/OData/odata.net/blob/cc9d86f0069be395a4f07aabe5cad0cec7b9a2cd/src/Microsoft.OData.Core/Json/NonIndentedTextWriter.cs#L47) in the synchronous workflow.

Reply:
@gathogojr maybe that should be mentioned in a comment?